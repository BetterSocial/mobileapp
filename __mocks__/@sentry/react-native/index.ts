export const addBreadcrumb = jest.fn();
