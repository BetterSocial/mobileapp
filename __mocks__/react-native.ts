import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN