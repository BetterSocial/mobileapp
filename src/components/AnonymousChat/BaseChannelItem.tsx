import * as React from 'react';
import {View} from 'react-native';

import ChannelImage from './child/ChannelImage';
import ChannelPostNotificationMessage from './child/ChannelPostNotificationMessage';
import ChannelPostNotificationStats from './child/ChannelPostNotificationStats';
import ChannelTitle from './child/ChannelTitle';
import CustomPressable from '../CustomPressable';
import styles from './BaseChannelItemStyles';
import {
  BaseChannelItemProps,
  BaseChannelItemTypeProps
} from '../../../types/component/AnonymousChat/BaseChannelItem.types';

const BaseChannelItem: (props: BaseChannelItemProps) => React.ReactElement = ({
  anonPostNotificationUserInfo = null,
  block = 0,
  comments = 0,
  downvote = 0,
  isCommentExists = false,
  isMe = false,
  message = '',
  name = '',
  picture = 'https://fastly.picsum.photos/id/173/400/400.jpg?hmac=TU_DMkn7FSRRwiEpVveLvmyATg-y4hNrNKT-Cs4VQ1M',
  postMaker = null,
  postNotificationMessageText = '',
  postNotificationMessageUser = null,
  postNotificationPicture = 'https://fastly.picsum.photos/id/912/400/400.jpg?hmac=Wg3Y7jTiQxHr_NpRsTrHG58kBfZQGNeH2tCl5u2Ipr0',
  time = '12:00 PM',
  type = BaseChannelItemTypeProps.ANON_PM,
  unreadCount = 0,
  upvote = 0,
  onPress = () => {
    console.log('onPress');
  }
}) => {
  return (
    <CustomPressable onPress={onPress}>
      <View style={styles.chatContainer}>
        <ChannelImage
          type={type}
          mainPicture={picture}
          postNotificationPicture={postNotificationPicture}
          anonPostNotificationUserInfo={anonPostNotificationUserInfo}
          isCommentExists={isCommentExists}
          postMaker={postMaker}
        />

        <View style={styles.chatContentContainer}>
          <ChannelTitle
            type={type}
            name={name}
            message={message}
            time={time}
            unreadCount={unreadCount}
            isMe={isMe}
          />

          {/* Post Notification Message */}
          <View style={styles.chatContentSection}>
            <ChannelPostNotificationMessage
              type={type}
              commenterName={postNotificationMessageUser}
              message={postNotificationMessageText}
            />
          </View>

          {/* Post Stats */}
          <View style={styles.chatContentSection}>
            <ChannelPostNotificationStats
              type={type}
              block={block}
              upvote={upvote}
              downvote={downvote}
              comments={comments}
            />
          </View>
        </View>
      </View>
    </CustomPressable>
  );
};

export default BaseChannelItem;
