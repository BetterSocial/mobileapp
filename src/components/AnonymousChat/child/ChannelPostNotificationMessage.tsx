import * as React from 'react';
import {StyleSheet, Text} from 'react-native';

import {BaseChannelItemTypeProps} from '../../../../types/component/AnonymousChat/BaseChannelItem.types';
import {COLORS} from '../../../utils/theme';
import {fonts, normalize} from '../../../utils/fonts';

const ChannelPostNotificationMessage = ({
  type = BaseChannelItemTypeProps.ANON_PM,
  commenterName,
  message
}) => {
  const styles = StyleSheet.create({
    chatContentPostNotificationMessage: {
      fontFamily: fonts.inter[400],
      fontSize: normalize(12),
      lineHeight: 20,
      color: COLORS.blackgrey,
      marginRight: 29
    },
    chatContentPostNotificationMessageBold: {
      fontFamily: fonts.inter[600],
      fontSize: normalize(12),
      lineHeight: 20,
      color: COLORS.blackgrey
    }
  });

  const isPM = type?.includes('PM');
  const amICommenter = type?.includes('I_COMMENTED');

  if (isPM) return <></>;

  if (!commenterName)
    return (
      <Text
        numberOfLines={2}
        ellipsizeMode="tail"
        style={styles.chatContentPostNotificationMessage}>
        <Text style={styles.chatContentPostNotificationMessageBold}>{'No comments yet'}</Text>
      </Text>
    );

  const commenter = amICommenter ? 'You' : commenterName;

  return (
    <Text numberOfLines={2} ellipsizeMode="tail" style={styles.chatContentPostNotificationMessage}>
      <Text
        style={styles.chatContentPostNotificationMessageBold}>{`${commenter} commented: `}</Text>
      {message}
    </Text>
  );
};

export default ChannelPostNotificationMessage;
