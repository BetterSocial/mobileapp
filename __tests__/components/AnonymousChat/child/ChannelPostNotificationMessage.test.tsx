import * as React from 'react';
import {render} from '@testing-library/react-native';

import ChannelPostNotificationMessage from '../../../../src/components/AnonymousChat/child/ChannelPostNotificationMessage';
import {BaseChannelItemTypeProps} from '../../../../types/component/AnonymousChat/BaseChannelItem.types';

jest.mock('../../../../src/hooks/core/profile/useProfileHook', () => ({
  __esModule: true,
  default: () => ({profile: {user_id: '123'}})
}));

beforeEach(() => {
  jest.clearAllMocks();
});

describe('TESTING ChatItemTargetText', () => {
  it('RENDER ANON_PM should match snapshot', () => {
    const {toJSON} = render(
      <ChannelPostNotificationMessage type={BaseChannelItemTypeProps.ANON_PM} />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('RENDER non ANON_PM with no commenter name should match snapshot', () => {
    const {toJSON} = render(
      <ChannelPostNotificationMessage type={BaseChannelItemTypeProps.ANON_POST_NOTIFICATION} />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('RENDER non ANON_PM with same profile id should match snapshot', () => {
    const {toJSON} = render(
      <ChannelPostNotificationMessage
        type={BaseChannelItemTypeProps.ANON_POST_NOTIFICATION}
        commenterId={'123'}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('RENDER non ANON_PM with different profile id should match snapshot', () => {
    const {toJSON} = render(
      <ChannelPostNotificationMessage
        type={BaseChannelItemTypeProps.ANON_POST_NOTIFICATION}
        commenterId={'456'}
        commenterName={'commenterName'}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
