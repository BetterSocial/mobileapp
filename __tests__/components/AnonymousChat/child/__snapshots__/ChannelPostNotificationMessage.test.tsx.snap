// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TESTING ChatItemTargetText RENDER ANON_PM should match snapshot 1`] = `null`;

exports[`TESTING ChatItemTargetText RENDER non ANON_PM with different profile id should match snapshot 1`] = `
<Text
  ellipsizeMode="tail"
  numberOfLines={2}
  style={
    Object {
      "color": "#7E8FA0",
      "fontFamily": "Inter-Regular",
      "fontSize": 10.8,
      "lineHeight": 20,
      "marginRight": 29,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#7E8FA0",
        "fontFamily": "Inter-SemiBold",
        "fontSize": 10.8,
        "lineHeight": 20,
      }
    }
  >
    commenterName commented: 
  </Text>
</Text>
`;

exports[`TESTING ChatItemTargetText RENDER non ANON_PM with no commenter name should match snapshot 1`] = `
<Text
  ellipsizeMode="tail"
  numberOfLines={2}
  style={
    Object {
      "color": "#7E8FA0",
      "fontFamily": "Inter-Regular",
      "fontSize": 10.8,
      "lineHeight": 20,
      "marginRight": 29,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#7E8FA0",
        "fontFamily": "Inter-SemiBold",
        "fontSize": 10.8,
        "lineHeight": 20,
      }
    }
  >
    No comments yet
  </Text>
</Text>
`;

exports[`TESTING ChatItemTargetText RENDER non ANON_PM with same profile id should match snapshot 1`] = `
<Text
  ellipsizeMode="tail"
  numberOfLines={2}
  style={
    Object {
      "color": "#7E8FA0",
      "fontFamily": "Inter-Regular",
      "fontSize": 10.8,
      "lineHeight": 20,
      "marginRight": 29,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#7E8FA0",
        "fontFamily": "Inter-SemiBold",
        "fontSize": 10.8,
        "lineHeight": 20,
      }
    }
  >
    No comments yet
  </Text>
</Text>
`;
